Documentation/cli-installation-scenarios.md

```diff
@@ -23,26 +26,99 @@ Obtaining .NET CLI
+
+## Installation process
+Each of the components listed in the previous section will have an installer/package. The dependencies of the installers between themseslves are given in the table below.
+
+| Installer 	| Depends on 	|
+|-----------	|------------	|
+| SDK       	| redist     	|
+| Redist    	| muxer      	|
+| muxer     	| -          	|
+
+The installation process will depend on the platform and the way of the install. For those installers that don't have automatic dependency resolution (Windows installer, OS X PKG) the installers will chain the installers of the components they depend on. DEB, RPM and similar will declare proper dependencies and the package manager will do the Right Thing(tm) by default. 
+
+From the table, we can see that if you install the SDK using `apt-get` for instance, you will get also a redist and a muxer. They will be separate packages, but will be declared as dependencies (similar for `yum`). Similar for the redist package. 
+
+The muxer is slightly a special case. Though there will be an installer, as mentioned in the previous section, it is an implementation detail. That means that acquiring the muxer should be done through either the SDK or the shared runtime installers. The only situation where this rule would not be true is if there was a major servicing event (e.g. a security update); in that case, the users would use the installer for the muxer directly, as we would rev its version accordingly. 
+
+The script installers are slightly different as they operate on zips/tarballs. The zip/tarball for the SDK will contain the entire set of things needed to be put on the disk.
+
+## Removal process
+Removing the bits from the machine **must** follow the order outlined above in installation. If the SDK is installed, it needs to be removed first and then the Redist and only then the muxer. Similar for the Redist.  
+
+## Upgrading the CLI 
+The semantics of installing the CLI will be side-by-side by default. This means that each new version will get installed besides any existing versions on disk. The [layout section](#layout-on-disk) covers how that will look like on disk.
+
+Since this is the case, there is no special "upgrade". When the user needs a new  version, the user just installs the new version using any of the installers specified in this document. The installer will just drop a new version at the predefined location. 
+
+This, however, does have one specific constraint: **newever versions must be installed in the same location the previous version was in**. This constraint is due to the fact that the "muxer" uses convention to figure out how to find the actual driver that the user reuqested. 
+
+## Layout on disk
+```
+~/dotnet
+	-dotnet (%PATH%)
+	-SDK
```

Comment:
agreed with @blackdwarf . I'd simply call it cli/dotnet-cli or something like that